app/forms/application_form.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class ApplicationForm
+  include ActiveModel::Naming
+  include ActiveModel::Validations
+
+  # Overrides <tt>ActiveModel::Naming#model_name</tt> to take <tt>resource_name</tt>
+  # into account.
+  #
+  # @return [ActiveModel::Name]
+  def self.model_name
+    ActiveModel::Name.new(self, nil, resource_class.name)
+  end
+
+  # Sets or infers the resource name.
+  #
+  # @param resource [String] the name of the resource class
+  # @return [String]
+  # @example A class where the form object name does not match the resource:
+  #   class SignupForm < ApplicationForm
+  #     resource_name User
```

Comment:
should this be `resource_class`?

Reply:
It should, a leftover from when I renamed the method, thanks for spotting this!